src/ixdat/exporters/csv_exporter.py

```diff
@@ -1,74 +1,128 @@
+        self.prepare_header_and_data(measurement, v_list, tspan)
+        self.prepare_column_header()
+        self.write_header()
+        self.write_data()
+
+    def prepare_header_and_data(self, measurement, v_list, tspan):
+        """Prepare self.header_lines to include metadata and value-time pairs
+
+        Args:
+            measurement (Measurement): The measurement being exported
+            v_list (list of str): The names of the ValueSeries to include
+            tspan (timespan): The timespan of the data to include in the export
+        """
+        columns_data = {}
+        s_list = []
+        v_list = v_list or self.default_v_list or list(measurement.value_names)
+
+        timecols = {}
         for v_name in v_list:
             t_name = measurement[v_name].tseries.name
             t, v = measurement.grab(v_name, tspan=tspan)
-            if t_name not in columns_data:
+            if t_name in timecols:
+                timecols[t_name].append(v_name)
+            else:
                 columns_data[t_name] = t
                 s_list.append(t_name)
+                timecols[t_name] = [v_name]
             columns_data[v_name] = v
             s_list.append(v_name)
 
-        header_line = (
-            "".join([s_name + self.delimiter for s_name in s_list])[
-                : -len(self.delimiter)
-            ]
+        header_lines = []
+        for attr in ["name", "technique", "tstamp", "backend_name", "id"]:
+            line = f"{attr} = {getattr(measurement, attr)}\n"
+            header_lines.append(line)
+        for t_name, v_names in timecols.items():
+            line = (
+                f"timecol '{t_name}' for: "
+                + " and ".join([f"'{v_name}'" for v_name in v_names])
+                + "\n"
+            )
+            header_lines.append(line)
+        if self.aliases:
+            aliases_line = f"aliases = {json.dumps(self.aliases)}\n"
+            header_lines.append(aliases_line)
+        self.header_lines = header_lines
+        self.s_list = s_list
+        self.columns_data = columns_data
+
+    def prepare_column_header(self):
+        """Prepare the column header line and finish the header_lines"""
+        N_header_lines = len(self.header_lines) + 3
+        self.header_lines.append(f"N_header_lines = {N_header_lines}\n")
+        self.header_lines.append("\n")
+
+        col_header_line = (
+            "".join([s_name + self.delim for s_name in self.s_list])[: -len(self.delim)]
             + "\n"
         )
+        self.header_lines.append(col_header_line)
 
-        lines = [header_line]
-        # All columns must have the length of the longest vector to keep data aligned:
-        max_length = max([len(data) for data in columns_data.values()])
+    def write_header(self):
+        """Create the file and write the header lines."""
+        with open(self.path_to_file, "w") as f:
+            f.writelines(self.header_lines)
+
+    def write_data(self):
```

Comment:
So the problem with pandas for sure, and also to the best of my knowledge, standard csv exporters, is that they require a rectangular data grid. i.e., every column has to be the same length. That will not generally be the case for ixdat measurements.

Reply:
Ahh, I see. I also noted further down in the review that pandas don't support 2 character separators which I think you use for the ixdat export format. It might be possible to hack the problem with the rectangular data grid, but let's leave that for later.